### DISCUSSION TITLE
KNX Monitor

### DISCUSSION THREAD

**Discussion by gisbertpeter** [brutella/hkknx-public#139] [General]
Hallo,
wenn man den KNX-Monitor nutzt und viel Bus-Traffic hat beendet sich das Monitoring scheinbar nach x Minuten, d.h. es kommen keine Ausschriften mehr. Das scheint auch Auswirkungen auf die Stabilität der Software zu haben. Gibt es Begrenzungen oder Einstellungen, die man beachten muss? Kann man überprüfen wie der Status der HKKNX Software ist?

**Comment by brutella** [comment]
> wenn man den KNX-Monitor nutzt und viel Bus-Traffic hat beendet sich das Monitoring scheinbar nach x Minuten, d.h. es kommen keine Ausschriften mehr.

Das Problem ist mir neu. Kannst du mal überprüfen wieviel Speicher der Prozess `hkknx` verbraucht, wenn der Gruppenmonitor aktiv ist.

> Kann man überprüfen wie der Status der HKKNX Software ist?

Unter Linux mit `top`, unter macOS mit der Aktivitätsanzeige.

**Reply by gisbertpeter** [reply]
@brutella 
Das ist der Zustand, wo zu HomeKit keine Connection zu HKKNX hat, nach ca. 30 Minuten Anzeige Gruppenmonitor und Systemlog.
Ich habe den Eindruck, dass das HKKNX Gruppenmonitoring Einfluß auf die Stabilität hat.

----- Memory ----------------------------
pi@homebridge:~ $ top
top - 12:45:44 up 23 min,  1 user,  load average: 0.07, 0.09, 0.12
Tasks: 136 total,   1 running, 135 sleeping,   0 stopped,   0 zombie
%Cpu(s):  3.2 us,  1.8 sy,  0.1 ni, 94.8 id,  0.0 wa,  0.0 hi,  0.1 si,  0.0 st
MiB Mem :   3827.3 total,   3105.3 free,    397.0 used,    325.0 buff/cache
MiB Swap:    100.0 total,    100.0 free,      0.0 used.   3370.0 avail Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
  482 pi        20   0  811436  24440  13124 S   4.6   0.6   0:31.95 hkknx

------ Memory hkknx -----------------
pi@homebridge:~ $ pmap -d 482
482:   /home/pi/hkknx -db /home/pi/database
Address   Kbytes Mode  Offset           Device    Mapping
00010000    6924 r-x-- 0000000000000000 0b3:00002 hkknx
006e0000    7756 r---- 00000000006d0000 0b3:00002 hkknx
00e80000     588 rw--- 0000000000e70000 0b3:00002 hkknx
00f13000     100 rw--- 0000000000000000 000:00000   [ anon ]
01800000   12288 rw--- 0000000000000000 000:00000   [ anon ]
02400000  516096 ----- 0000000000000000 000:00000   [ anon ]
a69e1000    3656 rw--- 0000000000000000 000:00000   [ anon ]
a6d73000  263616 ----- 0000000000000000 000:00000   [ anon ]
b6ee3000     268 rw--- 0000000000000000 000:00000   [ anon ]
befa9000     132 rw--- 0000000000000000 000:00000   [ stack ]
beff9000       4 r-x-- 0000000000000000 000:00000   [ anon ]
beffa000       4 r---- 0000000000000000 000:00000   [ anon ]
beffb000       4 r-x-- 0000000000000000 000:00000   [ anon ]
ffff0000       4 r-x-- 0000000000000000 000:00000   [ anon ]
mapped: 811440K    writeable/private: 17032K    shared: 0K

---- Konsole -------------------------------
DEBUG | 2021/12/23 12:37:59 | groupmon.go:139 | Unknown dpst DPT-16
DEBUG | 2021/12/23 12:37:59 | groupmon.go:139 | Unknown dpst DPST-9-6
DEBUG | 2021/12/23 12:38:02 | accessories.go:13 | 192.168.0.3:55516 GET /accessories
DEBUG | 2021/12/23 12:38:02 | connection.go:112 | Close connection and remove session
DEBUG | 2021/12/23 12:38:09 | groupmon.go:139 | Unknown dpst DPT-16
DEBUG | 2021/12/23 12:38:09 | groupmon.go:139 | Unknown dpst DPST-9-6
DEBUG | 2021/12/23 12:38:15 | connection.go:112 | Close connection and remove session
DEBUG | 2021/12/23 12:38:15 | connection.go:112 | Close connection and remove session
DEBUG | 2021/12/23 12:38:19 | groupmon.go:139 | Unknown dpst DPT-16
DEBUG | 2021/12/23 12:38:19 | groupmon.go:139 | Unknown dpst DPST-9-6
DEBUG | 2021/12/23 12:38:29 | groupmon.go:139 | Unknown dpst DPT-16
DEBUG | 2021/12/23 12:38:29 | groupmon.go:139 | Unknown dpst DPST-9-6
DEBUG | 2021/12/23 12:38:39

------KNX Gruppenmonitor --------------
ca. 20 Messages every 10 sec

**Reply by brutella** [reply]
Interessant wäre zu sehen, ob hkknx über die Zeit immer mehr Speicherplatz benötigt und deshalb irgendwann "hängt".
